BoxContentSDK/BoxContentSDK/AppToApp/BOXAppToAppMessage.m

```diff
@@ -333,13 +333,17 @@ - (id)initWithCoder:(NSCoder *)aDecoder
     if (self = [self init])
     {
         _messageType = [aDecoder decodeInt32ForKey:@"messageType"];
-        _senderApplication = [aDecoder decodeObjectForKey:@"senderApplication"];
-        _receiverApplication = [aDecoder decodeObjectForKey:@"receiverApplication"];
-        _action = [aDecoder decodeObjectForKey:@"action"];
-        _fileMetadata = [aDecoder decodeObjectForKey:@"metadata"];
+        _senderApplication = [aDecoder decodeObjectOfClass:[BOXAppToAppApplication class] forKey:@"senderApplication"];
+        _receiverApplication = [aDecoder decodeObjectOfClass:[BOXAppToAppApplication class] forKey:@"receiverApplication"];
+        _action = [aDecoder decodeObjectOfClass:[NSString class] forKey:@"action"];
+        _fileMetadata = [aDecoder decodeObjectOfClass:[BOXAppToAppFileMetadata class] forKey:@"metadata"];
```

Comment:
Does this class support secure coding?

Reply:
Not sure what you mean, can you please clarify a little more?

Reply:
if `BOXAppToAppFileMetadata` doesn't correctly implement secure decoding, then this class can't possibly be secure in its decoding.

Reply:
`BOXAppToAppFileMetadata` contains some NSString properties which are put into a dictionary using the two `appToAppFileMetadataWithFileName` methods to form the metadata. This metadata dictionary is what's encoded/decoded inside that class.

So I guess we can safely say `BOXAppToAppFileMetadata` is safely decoded or implements safe decoded?

Reply:
It must conform to `NSSecureCoding` and use the appropriate decoding methods, or it's not secure.

Reply:
Caught up offline. `BOXAppToAppFileMetadata` conforms to `NSSecureCoding` and implements the `supportsSecureCoding` method.